modules/pubgeniusBidAdapter.js

```diff
@@ -0,0 +1,195 @@
+      const consent = gdpr.consentString;
+      deepSetValue(data, 'regs.ext.gdpr', numericBoolean(applies));
+      if (applies && consent) {
+        deepSetValue(data, 'user.ext.consent', consent);
+      }
+    }
+
+    const usp = bidderRequest.uspConsent;
+    if (usp) {
+      deepSetValue(data, 'regs.ext.us_privacy', usp);
+    }
+
+    const schain = bidderRequest.schain;
+    if (schain) {
+      deepSetValue(data, 'source.ext.schain', schain);
+    }
+
+    if (config.getConfig('coppa')) {
+      deepSetValue(data, 'regs.coppa', 1);
+    }
+
+    const bidUserIdAsEids = deepAccess(bidRequests, '0.userIdAsEids');
+    if (bidUserIdAsEids && bidUserIdAsEids.length) {
+      const eids = bidUserIdAsEids.filter(eid => eid.source === 'adserver.org');
+      if (eids.length) {
+        deepSetValue(data, 'user.ext.eids', eids);
+      }
+    }
+
+    return {
+      method: 'POST',
+      url: AUCTION_URL,
+      data,
+      options: { contentType: 'application/json' },
+    };
+  },
+
+  interpretResponse({ body }) {
+    const bidResponses = [];
+    const currency = body.cur || 'USD';
+    const seatbids = body.seatbid;
+    if (seatbids) {
+      seatbids.forEach(seatbid => {
+        seatbid.bid.forEach(bid => {
+          const bidResponse = interpretBid(bid);
+          bidResponse.currency = currency;
+          bidResponses.push(bidResponse);
+        });
+      });
+    }
+    return bidResponses;
+  },
+
+  getUserSyncs(syncOptions, serverResponses, gdprConsent, uspConsent) {
+    const syncs = []
+
+    if (syncOptions.iframeEnabled) {
+      let params = {};
+      if (gdprConsent) {
+        params.gdpr = numericBoolean(gdprConsent.gdprApplies);
+        if (gdprConsent.consentString) {
+          params.consent = gdprConsent.consentString;
+        }
+      }
+      if (uspConsent) {
+        params.us_privacy = uspConsent;
+      }
+
+      const qs = parseQueryStringParameters(params);
+      syncs.push({
+        type: 'iframe',
+        url: `${BASE_URL}/usersync/pixels.html?${qs}`,
+      });
+    }
+
+    return syncs;
+  },
+
+  onTimeout(data) {
+    ajax(`${BASE_URL}/prebid/events?type=timeout`, null, JSON.stringify(data), {
+      method: 'POST',
+      contentType: 'application/json',
+    });
+  },
+};
+
+function buildImp(bid) {
+  const imp = {
+    id: bid.bidId,
+    banner: {
+      format: deepAccess(bid, 'mediaTypes.banner.sizes').map(size => ({ w: size[0], h: size[1] })),
+      topframe: numericBoolean(!inIframe()),
+    },
+    tagid: String(bid.params.adUnitId),
+  };
+  const bidFloor = bid.params.bidFloor;
+  if (isNumber(bidFloor)) {
+    imp.bidfloor = bidFloor;
+  }
+  const pos = bid.params.position;
+  if (isInteger(pos)) {
+    imp.banner.pos = pos;
+  }
+  return imp;
+}
+
+function buildSite(bidderRequest) {
+  const pageUrl = config.getConfig('pageUrl') || bidderRequest.refererInfo.referer;
+  if (pageUrl) {
+    return {
+      page: encodeURIComponent(pageUrl),
```

Comment:
Why encode it?

Reply:
Good catch! I don't think it's necessary. I copied some lines from other bidders and thought I'd come back later to see why they had encoding but then forgot.

I'll remove it.